### PR TITLE
VMware: add support for customize existing VM directly

### LINKED ISSUE #51010: VMware: request support applying OS customization to existing VM
##### SUMMARY

vmware_guest can only be used for clone a new VM from template/existing VM and apply existing customization specification to the cloned VM. We want to create a new customization spec firstly, and then apply this spec to the existing VM to customize the Guest OS.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

vmware_guest_customization
##### ADDITIONAL INFORMATION

(1) can used to create new customization specifications for Linux or Windows guest OS.
(2) apply created customization spec to existing virtual machine to customize guest OS.

```yaml
e.g., 
customize_spec: new
customize_spec_name: win_spec_test
customize_spec_org: gosc_test
customize_spec_os: windows
.... (parameters required to create a windows customize spec)
```

### PR DESCRIPTION
**Pull request by Tomorrow9**
##### SUMMARY

Fixes #51010
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

vmware_guest
##### ADDITIONAL INFORMATION

Add a new parameter "existing_vm" in "customization" to specify that OS customization will be applied to the specified VM, not cloned from the template or VM.
- name: "customize the specified VM"
      vmware_guest:
        validate_certs: False
        hostname: 10.10.10.10
        username: Administrator@vsphere.local
        password: test
        folder: ""
        name: dw_test_vm
        state: present
        customization:
          existing_vm: True
          autologon: True
          autologoncount: 1
          password: test123
          runonce:
            - powershell.exe mkdir C:\users\test
        networks:
          - name: VM Network
            type: static
            ip: 192.168.22.22
            netmask: 255.255.255.0
            gateway: 192.168.1.1
            dns_servers: 192.168.1.2
        # customization_spec: dw_win
        wait_for_customization: True

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 1]
```diff
@@ -2491,14 +2529,24 @@ def main():
...
         elif module.params['state'] == 'present':
-            if module.check_mode:
-                result.update(
-                    vm_name=vm.name,
-                    changed=True,
-                    desired_operation='reconfigure_vm',
-                )
-                module.exit_json(**result)
-            result = pyv.reconfigure_vm()
+            if 'existing_vm' in module.params['customization'] and module.params['customization']['existing_vm']:
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py]: This means (in combination with line 2541-2549), that a VM can not be reconfigured and customized at the same time with this change?

**Review by pdellaert — COMMENTED** [review]
Overall great addition, have one question on combination of reconfigure and customization at the same time

File: lib/ansible/modules/cloud/vmware/vmware_guest.py (hunk 1 above)
Comment (Tomorrow9) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: Yeah, what I intended to do is "customize" and "re-configure" are two separate operations. You mean do "re-configure" first and then do "customize" on this VM? Yeah, that's also a scenario.

Comment (pdellaert) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: yeah, indeed... if it's not supported, it should probably be mentioned in the documentation?

Comment (Tomorrow9) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: I think these two operations can be executed in order if VM is in powered off state, or "customize" will fail. So what do you think put this "customize" operation at the end of "re-configure"? Thanks a lot.

Comment (pdellaert) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: I think that would make sense, but it's up to you, of course :).

Comment (Tomorrow9) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: Ok, thank you very much :), I'll make some changes on this later.

Comment (Tomorrow9) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: I moved the "customize existing vm" execution to the last part of "reconfigure vm".

**Review by pdellaert — APPROVED** [review]
LGTM

Thanks for the addition and the change to support customization of an existing VM :)

**Comment by L1ghtman2k** [comment]
Would it be possible to edit 'netwoks' with this flag?

**Pull request merged**